### ISSUE TITLE
Use of IQDropDownTextField

### ISSUE THREAD

**Issue by jonathanfontaine** [hackiftekhar/IQKeyboardManager#207]
Hi,

When I used a IQDropDownTextField, I set its mode like this :

    `myIqDropDownTextField.dropDownMode = IQDropDownModeDatePicker;`
    `myIqDropDownTextField.datePickerMode = UIDatePickerModeDateAndTime;`
When I touch the textField, the date picker appears as expected :
* If I pick **a different date than the default selected date**, the textField is updated, as expected ;
* But if I just touch OK, **without picking a date** (because the default selected date is fine for me), the textField is not filled.

In other words, the OK button just closes the picker, but it would be great that it fills the textField with the selected date in the picker.

**Comment by hackiftekhar** [comment]
What if user unexpectedly open date picker and don't want to choose date for some reason? That's why done button just resign first responder(For IQKeyboardManager done button work is to resign current textField only, library support Automatic toolbar creation but doesn't forcefully support 'IQDropDownTextField'). But you can add customised done selector defined in 'IQUIView+IQKeyboardToolbar .h', so you can do you customised work on done button action.

**Comment by jonathanfontaine** [comment]
OK, I didn't think about that. But the keyboard can be resigned if we touch anywhere else in the view. Plus, if I want the default selected date, I'm obliged to change it to another date and go back on the first date, it's not user friendly and hard to understand.

**Comment by hackiftekhar** [comment]
That's why I left this option to developer. You can also implement textFieldDidEndEditing and you can do your customised logic there. Try to understand that every app and condition has their own logic, there is no standard on force select selection on textField. May be you want force selection, but some other app has different condition and developer does't want force selection. That's why I don't force select. But you have the rights to implement textFieldDidEndEditing and write your logic. Let me know if you need more clarification or any help.

**Comment by hackiftekhar** [comment]
May be I need to implement 'isOptional' boolean of 'IQDropDownTextField' for date and time icoer too for force selection. Right?

**Comment by jonathanfontaine** [comment]
That's OK, I understand ! Thanks.

**Comment by hackiftekhar** [comment]
It's already implemented but only for textPicker mode(Not implemented for time/date picker).

**Comment by hackiftekhar** [comment]
Fixed 'isOptionalDropDown' boolean for dare/time picker on IQDropDownTextField repo.

**Comment by hackiftekhar** [comment]
Marking it as `bug` and `wontfix` because it's not relevant to 'IQKeyboardManager' and fixed in 'IQDropDownTextField'.

**Comment by jonathanfontaine** [comment]
OK. I tested your code, it works fine ! Thanks.

**Comment by hackiftekhar** [comment]
Thanks :)
